src/snapred/backend/dao/ingredients/DiffractionCalibrationIngredients.py

```diff
@@ -16,3 +16,10 @@ class DiffractionCalibrationIngredients(BaseModel):
     groupedPeakLists: List[GroupPeakList]
     convergenceThreshold: float
     maxOffset: float = Config["calibration.diffraction.maximumOffset"]
+    units: str = "TOF"
+
+    @validator("units")
+    def validatedSpacing(cls, v):
+        if v not in ["TOF", "dSpacing"]:
+            raise ValueError(f"dSpacing must be either 'TOF' or 'dSpacing', not '{v}'")
+        return v
```

Comment:
For future reference, the simpler way to do this is with the `Literal` type, as in
``` python
units: Literal["TOF", "dSpacing"]
```
with no further validators needed.

Reply:
Thanks for this tip! Guess I didn't think about doing it this way, but this is much better than creating validation logic.